lib/es-models/base.js

```diff
@@ -40,9 +39,31 @@ class EsBase {
    *  `EsItem` instances, calling `.toJson` on the `EsBib` instance will
    *  trigger calling `.toJson` on each of the `EsItem` instances.
    */
-  toJson () {
-    // Apply to-json to this object:
-    return toJson(this)
+  toPlainObject (schema) {
+    const properties = Object.keys(schema)
+    const plainObject = {}
+    properties.forEach((property) => {
+      if (!this[property]) {
+        logger.warn(`No method on ${this.constructor.name} model for schema property ${property}`)
+        return
+      }
+      // if the schema property correspond to holdings or items
+      const isChildMapping = !!schema[property]?.properties?.uri
+      let value
+      try {
+        value = this[property]()
+      } catch (e) {
+        logger.warn(`Error calling ${this.constructor.name}.${property}: `, e)
+      }
+      if ((!value)) {
+        logger.debug(`Skipping ${property} because value is ${value}`)
+        return
+      }
+      if (isChildMapping) {
+        plainObject[property] = value?.map((nestedModel) => nestedModel.toPlainObject(schema[property].properties))
```

Comment:
The `toJson` method used the existence of a `toJson` method on the child instance to determine whether to recurse. Since you're using "is-child" to determine whether or not to call `toPlainObject` on the child object, couldn't the existence of the special `toPlainObject` also serve to identify children? I think the current uri check is fine - just pointing out an alternative that may be a more durable check (i.e. future es-models may not define a `uri`, but they definitely have to implement `toPlainObject`).